src/labelle/lib/devices/device_config.py

```diff
@@ -50,19 +55,57 @@ def matches_device_id(self, idValue: int) -> bool:
         else:
             return False
 
-    def get_tape_print_margins(self, tapeSizeMM: int) -> tuple[int, int]:
+    def get_tape_print_size_and_margins_px(
+        self, tapeSizeMM: int
+    ) -> tuple[int, int, int]:
         """Get print margins for a specific tape size.
 
         :param tapeSizeMM: Tape size in mm
-        :return: Margins tuple, None if not supported
+        :return: Margins tuple in pixels [ActivePixels,TopMarginPx,BottomMarginPx]
         """
         if tapeSizeMM in self.supportedTapeSizes:
-            if tapeSizeMM in self.marginsPerTape:
-                # Return specified margins
-                return self.marginsPerTape[tapeSizeMM]
+            # Calculate the pixels per mm for this printer
+            # Example: printhead of 128 Pixels, distributed over 18 mm of active area.
+            #   Makes 7.11 pixels/mm
+            printPPmm: float = (
+                self.printHeadSizePixels / self.printHeadActiveAreaMillimeters
+            )
+
+            # Calculate usable tape width (*2 for top and bottom)
+            usableTapeWidth: float = tapeSizeMM - (
+                self.TAPE_ALIGNMENT_INACCURACY_MM * 2
+            )
+
+            # Calculate the numer of active pixels for the tape
+            activeTapePixels: float = 0
+            if usableTapeWidth >= self.printHeadActiveAreaMillimeters:
+                # Tape is larger than active area of printhead. Use all pixels
+                activeTapePixels = self.printHeadSizePixels
             else:
-                # No specific margins set, return default
-                return (0, 0)
+                # Calculate the amount of active pixels we are able to use
+                # (taking the placement inaccuracy into account)
+                activeTapePixels = printPPmm * usableTapeWidth
+
+            # Round down to nearest whole number as we can't use half a pixels ;)
+            activeTapePixels = math.floor(activeTapePixels)
+
+            # To calculate the margins we need to know some hardware info
+            # Printer has special "support studs" that
+            # let 19 & 24mm tapes go to the bottom (if supported)
+            # but 12mm based casettes are raised so they are centered to the printhead.
+            # Smaller tapes than 12mm are in 12mm casettes are centered
+            # in the cartridge and in turn also centered to the printhead
+            # Which gives us the advantage that we can
+            # just calculate the top and bottom margin
+
+            # Calculate the top margin
+            topMargin = round(((self.printHeadSizePixels - activeTapePixels) / 2), 0)
+
+            # Bottom margin is equal due to centering of the tape
+            bottomMargin = topMargin
+
+            # Return active pixels / margins set
+            return (int(activeTapePixels), int(topMargin), int(bottomMargin))
         else:
             # Tape size not supported
             raise ValueError(
```

Comment:
Here we encounter handling of branch from 60 lines ago... so instead, why not in line 66:
```
if tapeSizeMM not in self.supportedTapeSizes:
    raise ...
```

BTW I know this is not your code, but I comment on everything that I notice while reading your changes, which also include existing code. I hope you are OK with that.

Reply:
I'm OK with that, and while I'm add it, I'll try to fix it.
But keep in mind that I'm a python n00b. So my strategy is: don´t touch things, you might break it...

Reply:
Fixed